h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -338,7 +338,8 @@ public void computeCrossValidation() {
 
       _job.setReadyForView(true);
       DKV.put(_job);
-
+    } catch (Exception e) {
+      DKV.remove(_job._result); //ensure there's no incomplete model left for manipulation after crash or cancellation
```

Comment:
This will only remove the object from DKV but will not call remove() on the object - is that intended

Reply:
no, should probably call _job.remove() instead. thanks

Reply:
by the way, also forgot to propagate the exception here